codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/BuilderGenerator.kt

```diff
@@ -258,6 +275,42 @@ class BuilderGenerator(
                 withBlock("$memberName: self.$memberName", ",") {
                     // Write the modifier
                     when {
+                        !memberSymbol.isOptional() && member.hasTrait<DefaultTrait>() -> {
+                            val node = member.expectTrait<DefaultTrait>().toNode()!!
+                            when (val target = model.expectShape(member.target)) {
+                                is EnumShape, is IntEnumShape -> {
+                                    val value = when (target) {
+                                        is IntEnumShape -> node.expectNumberNode().value
+                                        else -> node.expectStringNode().value
+                                    }
+                                    val enumValues = when (target) {
+                                        is IntEnumShape -> target.enumValues
+                                        is EnumShape -> target.enumValues
+                                        else -> mapOf<String, String>()
+                                    }
+                                    val variant = enumValues
+                                        .entries
+                                        .filter { entry -> entry.value == value }
+                                        .map { entry -> symbolProvider.toEnumVariantName(EnumDefinition.builder().name(entry.key).value(entry.value.toString()).build())!! }
+                                        .first()
+                                    val symbol = symbolProvider.toSymbol(target)
+                                    rust(".unwrap_or($symbol::${variant.name})")
+                                }
+                                is IntegerShape, is FloatShape -> rust(".unwrap_or(${node.expectNumberNode().value})")
+                                is BooleanShape -> rust(".unwrap_or(${node.expectBooleanNode().value})")
+                                is StringShape -> rust(".unwrap_or(String::from(${node.expectStringNode().value.dq()}))")
```

Comment:
Recommend using `unwrap_or_else` to avoid unnecessary allocations.

Reply:
Thanks!